Slim/Plugin/MusicMagic/Plugin.pm

```diff
@@ -867,17 +873,22 @@ sub getMix {
 		return undef;
 	}
 
-	main::DEBUGLOG && $log->debug("Creating mix for: $validMixTypes{$for} using: $id as seed.");
-
-	if (!main::ISWINDOWS && ($validMixTypes{$for} eq 'song' || $validMixTypes{$for} eq 'album') ) {
-
-		# need to decode the file path when a file is used as seed
-		$id = Slim::Utils::Unicode::utf8decode_locale($id);
+	my $mixArgs = '';
+	my $count = scalar(@ids);
+	for (my $j = 0; $j < $count; $j++) {
```

Comment:
In Perl you can simplify this to:

```perl
foreach my $id (@ids) {
...
}
```

Or you use `map` to create the `$mixArgs`. Something like this:

```perl
my $mixArgs = join('&', map {
      my $id = main::ISWINDOWS ? $id : Slim::Utils::Unicode::utf8decode_locale($id);
      $validMixTypes{$for} . '=' . Plugins::MIPMixer::Common::escape($id);
} @ids);
```

`map` would create a new list based on `@ids`, and `join()` would concatenate its items using the `&` sign.